Life.java

```diff
@@ -91,7 +83,9 @@ public ILife nextGeneration() {
     return nGen;
   }
 
- 
+ public int getNeighbors(int x, int y){
+   return(Neighbors[x][y]);
+ }
 
   public void calculateNeighbors(){
```

Comment:
Diese Funktion, die die Adjazenzmatrix berechnet ist einerseits "genial" programmiert, andererseits für Normalsterbliche gerade noch nachvollziehbar und das wichtigste: sie funktioniert.

Dennoch ist sie nicht besonders schön. Ich würde das in einem Code-Review nur in extremen Ausnahmefällen (z.B. Performancegründe) durchgehen lassen.

Teilen Sie die Verantwortlichkeiten auf! 
In einer Integer Matrix speichern Sie ja die Anzahl der lebenden Nachbarn an jeder Stelle. 
Sie könnten die Berechnung der Nachbarn für eine Zelle in eine eigene Funktion auslagern.
Ich habe das vor ein paar Jahren mal in einem Coding Kata gemacht, da sehen Sie, wie das gemeint sein könnte:

```
public void calculateAdjazentMatrix( IntegerMatrix adjMatrix ) {
		int sum = 0;
		for( int x = 0; x < iMatrix.getDimensionX(); x++ ) {
			for( int y = 0; y < iMatrix.getDimensionY(); y++ ) {
				sum = calculateAdjSum( x, y, 1 );
				adjMatrix.set( x, y, sum );
			}
		}		
	}

private int calculateAdjSum( int x, int y, int distance ) {
		int minX = x - distance;
		int maxX = x + distance;
		int minY = y - distance;
		int maxY = y + distance;
		
		if( minX < 0 ) minX = 0;
		if( maxX >= iMatrix.getDimensionX() ) maxX = iMatrix.getDimensionX()-1;
		if( minY < 0 ) minY = 0;
		if( maxY >= iMatrix.getDimensionY() ) maxY = iMatrix.getDimensionY()-1;
		
		int sum = 0;
		for( int ix = minX; ix <= maxX; ix++ ) {
			for( int iy = minY; iy <= maxY; iy++ ) {
				sum += iMatrix.get(ix, iy);
			}
		}
		sum -= iMatrix.get( x, y );		
		return sum;
	}
```